app/controllers/api/notify_callbacks_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class Api::NotifyCallbacksController < Api::ApiController
+  def create
+    return head :no_content unless params[:reference] || params[:id]
+
+    if params[:reference]
+      email = NominationEmail.find_by(token: params[:reference])
+      email.update!(notify_status: params[:status]) if email
+    else
+      email = PartnershipNotificationEmail.find_by(notify_id: params[:id])
+      email.update!(notify_status: params[:status], delivered_at: params[:sent_at]&.to_datetime) if email
```

Comment:
This logic looks like a mess. I think it would be nice to have a reference that's decoupled from the things in the email. Kind of like the partnership emails do it.

Can we do it for Nomination emails too? As in, have a notify id?

Can we have delivered_at on both? 

In fact, do we want to have some inheritance between them?

Reply:
I agree best to make the two emails work in a similar way and using the notify_id as a reference is definitely better. I'd be happy to have that done in another pr though.

Reply:
Will pick up as part of this ticket, but in a new PR

Reply:
see https://github.com/DFE-Digital/early-careers-framework/pull/259